Dockerfile

```diff
@@ -1,33 +1,36 @@
-FROM hexpm/elixir:1.10.3-erlang-22.3.3-debian-stretch-20200224
+###
+# Three-stage Dockerfile
+###
 
-WORKDIR /root
+# 1.) Get the Elixir dependencies within an Elixir container
+FROM hexpm/elixir:1.10.3-erlang-22.3.3-debian-stretch-20200224 as elixir-builder
 
-ARG SENTRY_DSN=""
+# Install and configure Git to use HTTPS in order to avoid issues with the internal MBTA network
+RUN apt-get update && apt-get install -y git
+RUN git config --global url.https://github.com/.insteadOf git://github.com/
 
-# Debian dependencies
-RUN apt-get update && apt-get install -y curl git make build-essential
+WORKDIR /root
+ADD . .
 
-# Install node/npm
-# Instructions from https://nodejs.org/en/download/package-manager/#debian-and-ubuntu-based-linux-distributions
-RUN curl -sL https://deb.nodesource.com/setup_14.x | bash - && \
-  apt-get install -y nodejs
+# Install Hex+Rebar
+RUN mix local.hex --force && \
+  mix local.rebar --force
 
-# Clean up
-RUN apt-get clean
+RUN mix do deps.get --only prod
 
-ENV MIX_ENV=prod
 
-# Configure Git to use HTTPS in order to avoid issues with the internal MBTA network
-RUN git config --global url.https://github.com/.insteadOf git://github.com/
+# 2.) Build the frontend assets within a node.js container instead of installing node/npm
 
-# Install Hex+Rebar
-RUN mix local.hex --force && \
-  mix local.rebar --force
+# The Elixir image was built with debian stretch, use the node debian-stretch base
+FROM node:14.15.1-stretch as assets-builder
+
+ARG SENTRY_DSN=""
```

Comment:
This might be out of scope for this change, but why do we need to give this an empty value at build time? I see it's referenced on [line 37](https://github.com/mbta/dotcom/pull/693/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R37)... is this a workaround to some build-time configuration requirement for the Sentry JS library?

Reply:
This was already in the original dotcom `Dockerfile`. I don't know enough to attest that this is needed to have a working integration with Sentry.
Is "I can remove it and see if we get yelled at in prod" an option?

Reply:
> Is "I can remove it and see if we get yelled at in prod" an option?

Maybe, but it might be worth investigating how that env var is used first. In which case I would definitely call it out of scope for this change, but perhaps worth tasking out separately if the Sentry integration for JavaScript is something y'all care about. (Or ripping out entirely if you don't.) FWIW, Skate has a working Sentry integration for its frontend, so it might be worth looking at their implementation for reference.